### PR TITLE
Expose inlay hints to Razor cohosting

### PR DESCRIPTION
**Pull request by davidwengier**
Part of https://github.com/dotnet/razor/issues/9519

Also exposes a test helper for better Razor tests

### CODE REVIEW

File: src/Workspaces/Remote/ServiceHub/ExternalAccess/Razor/Api/TestHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor.Api
+{
+    internal static class TestHelpers
+    {
+        /// <summary>
+        /// Allows tests to create a new remote workspace, rather than use RemoteWorkspaceManager.Default, so things can be isolated
+        /// </summary>
+        public static Workspace CreateTestWorkspace()
+            => new RemoteWorkspaceManager(workspace => new SolutionAssetCache(workspace, cleanupInterval: TimeSpan.FromSeconds(30), purgeAfter: TimeSpan.FromMinutes(1))).GetWorkspace();
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/ServiceHub/ExternalAccess/Razor/Api/TestHelpers.cs:16]: don't love these hardcoded lceanups.

**Review by CyrusNajmabadi — APPROVED** [review]
signing off.  but DavidB should def look :)

File: src/Workspaces/Remote/ServiceHub/ExternalAccess/Razor/Api/TestHelpers.cs (hunk 1 above)
Comment (davidwengier) [reply src/Workspaces/Remote/ServiceHub/ExternalAccess/Razor/Api/TestHelpers.cs:16]: If it makes you feel any better, they're never used, and nor is the solution asset cache, in our tests anyway.

Comment (davidwengier) [reply src/Workspaces/Remote/ServiceHub/ExternalAccess/Razor/Api/TestHelpers.cs:16]: On second thought, I'm just going to delete this file. I didn't end up needing it now, so better to leave it for later

**Comment by davidwengier** [comment]
> signing off. but DavidB should def look :)

Thank you Cyrus. @dibarbet you have been summoned :)

**Review by dibarbet — APPROVED** [review]

**Pull request merged**